engine/src/main/java/io/camunda/zeebe/engine/processing/ProcessEventProcessors.java

```diff
@@ -225,12 +226,23 @@ private static void addProcessInstanceBatchStreamProcessors(
       final TypedRecordProcessors typedRecordProcessors,
       final MutableProcessingState processingState,
       final Writers writers) {
-    final TerminateProcessInstanceBatchProcessor terminateBatchProcessor =
+    final var terminateBatchProcessor =
         new TerminateProcessInstanceBatchProcessor(
             writers, processingState.getKeyGenerator(), processingState.getElementInstanceState());
     typedRecordProcessors.onCommand(
         ValueType.PROCESS_INSTANCE_BATCH,
         ProcessInstanceBatchIntent.TERMINATE,
         terminateBatchProcessor);
+
+    final var activateBatchProcessor =
+        new ActivateProcessInstanceBatchProcessor(
+            writers,
+            processingState.getKeyGenerator(),
+            processingState.getElementInstanceState(),
+            processingState.getProcessState());
+    typedRecordProcessors.onCommand(
+        ValueType.PROCESS_INSTANCE_BATCH,
+        ProcessInstanceBatchIntent.ACTIVATE,
+        activateBatchProcessor);
```

Comment:
:upside_down_face: You can chang the `typedRecordProcessors.onCommand` methods and instantiate the `*BatchProcessor` classes inside the method. We use that pattern in some of the other methods above (e.g. lines 160-170).

It looks nicer from my POV, but feel free to ignore it.